docs/src/modules/spring/pages/views.adoc

```diff
@@ -1 +1,169 @@
+
+ .src/main/java/customer/view/CustomerByNameView.class
+[source,java]
+----
+include::example$spring-eventsourced-customer-registry/src/main/java/customer/view/CustomerByNameView.java[tags=class]
+----
+<1> setting view ID 
+<2> updating the state of the view with the new name 
+
+An Event Sourced entity can emit many types of events. You need to define a method for each event type. They return an `UpdateEffect`, which describes next processing actions, such as updating the view state.
+
+
+
+
+See <<#query>> for more examples of valid query syntax.
+
+== Ignoring events
+
+When consuming events, each event must be matched by a View service method. In case your View is only interested in certain events:
+1. You can declare a catch-all method to receive the `sealed interface` and don't update the View.
+2. You can add `ignoreUnknown = true` to your `@Subcribe` annotation but only if it is a type level annotation. 
+If an event type can't be handled and there is no `ignoreUnknown = true` at type level, the View will fail. Views are designed to restart, but since the handler is missing, it will fail again.
+
+
+[#topic-view]
+== Creating a View from a topic
+
+The source of a View can be a topic. It works the same way as shown in <<event-sourced-entity>> or <<value-entity>>, but you define it with `@Subscribe.Topic` instead.
+
+
+=== Relational projection
+
+Instead of using `SELECT *` you can define what columns that will be used in the response message:
+
+
+In a similar way, you can include values from the request message in the response, for example `:request_id`:
+
+[source,proto,indent=0]
+----
+SELECT :request_id, customer_id as id, name FROM customers WHERE name = :customer_name
+----
+
+
+[#changing]
+== How to modify a View
+
+Kalix creates indexes for the View based on the query. For example, the following query will result in a View with an index on the `name` column:
+
+[source,proto,indent=0]
+----
+SELECT * FROM customers WHERE name = :customer_name
+----
+
+You may realize after a deployment that you forgot adding some parameters to the query Parameters that aren't exposed to the endpoint of the View. After adding these parameters the query is changed and therefore Kalix needs to add indexes for these new columns. For example, changing the above query to filter to add users that are active would mean that Kalix needs to build a View with the index on the `is-active` column.
+
+[source,proto,indent=0]
+----
+SELECT * FROM customers WHERE name = :customer_name AND is-active = true
+----
+
+Such changes require you to define a new View. Kalix will then rebuild it from the source event log or value changes.
+
+WARNING: Views from topics cannot be rebuilt from the source messages, because it might not be possible to consume all events from the topic again. The new View is built from new messages published to the topic.
```

Comment:
This is probably a problem for the other SDKs as well. True when we had only PubSub, but with Kafka it might be possible. It will depend on the retention criteria.

Reply:
roger that